components/reporters/slack/internal/reporter/issue.go

```diff
@@ -0,0 +1,190 @@
+package reporter
+
+import (
+	"bytes"
+	_ "embed"
+	"strings"
+	"text/template"
+
+	"github.com/go-errors/errors"
+	vf "github.com/smithy-security/smithy/sdk/component/vulnerability-finding"
+	ocsffindinginfo "github.com/smithy-security/smithy/sdk/gen/ocsf_ext/finding_info/v1"
+	ocsf "github.com/smithy-security/smithy/sdk/gen/ocsf_schema/v1"
+	"google.golang.org/protobuf/encoding/protojson"
+
+	"github.com/smithy-security/smithy/components/reporters/slack/internal/reporter/util"
+)
+
+//go:embed issue.tpl
+var issueTpl string
+
+type IssueData struct {
+	Description      string
+	Title            string
+	FindingID        uint64
+	FindingLink      string
+	TargetName       string
+	TargetLink       string
+	IsRepository     bool
+	IsPurl           bool
+	Confidence       string
+	CWE              string
+	CWELink          string
+	CVE              string
+	Tool             string
+	RunName          string
+	RunLink          string
+	FindingPath      string
+	FindingStartLine uint32
+	FindingEndLine   uint32
+	Reference        string
+}
+
+func (r slackReporter) getMsgs(findings []*vf.VulnerabilityFinding) ([]string, error) {
+	tpl, err := template.New("issue").Parse(issueTpl)
+	if err != nil {
+		return nil, errors.Errorf("could not parse thread template: %w", err)
+	}
+
+	const (
+		unknownValue = "unknown"
+		unsetValue   = "-"
+	)
+
+	var msgs []string
+	for _, finding := range findings {
```

Comment:
I would rather we move this logic into a constructor for the `IssueData` struct and then have a String method that uses the template to generate the message